src/EditorFeatures/Core/Diagnostics/AbstractPushOrPullDiagnosticsTaggerProvider.PushDiagnosticsTaggerProvider.cs

```diff
@@ -137,6 +137,14 @@ protected sealed override ITaggerEventSource CreateEventSource(ITextView? textVi
             var suppressedDiagnosticsSpans = (NormalizedSnapshotSpanCollection?)null;
             buffer?.Properties.TryGetProperty(PredefinedPreviewTaggerKeys.SuppressDiagnosticsSpansKey, out suppressedDiagnosticsSpans);
 
+            // This may look confusing, as we are the "PushDiagnosticTaggerProvider" but we're potentially get push vs.
+            // pull diagnostic buckets here.  The reason for this is that the DiagnosticMode being discussed here is
+            // whether or not we're in lsp-pull or push mode, not if tagging is in push/pull.  So, for example, we may
+            // be in LSP pull mode.  In that case, classification/squiggles/suggestions will all be disabled since LSP
+            // takes it over.  However, inline-diagnostics is still running in non-LSP, and it may be in either tagging
+            // push/pull mode.  If it's tagging-push mode, we will reach here, but then call GetPullDiagnosticBuckets
+            // since we're in LSP pull mode.  If it didn't call GetPullDiagnosticBuckets, it would get nothing back
+            // since GetPushDiagnosticBuckets returns nothing with LSP pull diagnostics is on.
```

Comment:
I wonder if the naming is the most confusing here.  Both feature flags are named pull/push.  Perhaps we should keep the pull/push tagging flag, rename the LSP one to be something like LSP on/off, and then rename the `GetPullDiagnosticBuckets`/`GetPushDiagnosticBuckets` to be related to the LSP flag instead (what it uses today)

Because the function of both methods is the same, they just assert whether the calling method matches the diagnostic option.

That might be a change for a separate PR though

Reply:
will do in followup! :)